scipy/linalg/basic.py

```diff
@@ -145,83 +162,97 @@ def solve(a, b, sym_pos=False, lower=False, overwrite_a=False,
         else:
             b1 = b1[:, None]
         b_is_1D = True
-    elif b1.ndim > 2:
-        b_is_ND = True
 
     r_or_c = complex if np.iscomplexobj(a1) else float
 
-    if assume_a in ('gen', 'sym', 'her', 'pos'):
-        _structure = assume_a
-    else:
+    # Backwards compatibility - old keyword.
+    if sym_pos:
+        assume_a = 'pos'
+
+    if assume_a not in ('gen', 'sym', 'her', 'pos'):
         raise ValueError('{} is not a recognized matrix structure'
                          ''.format(assume_a))
 
     # Deprecate keyword "debug"
     if debug is not None:
         warnings.warn('Use of the "debug" keyword is deprecated '
-                      'and this keyword will be removed in the future '
+                      'and this keyword will be removed in future '
                       'versions of SciPy.', DeprecationWarning)
 
-    # Backwards compatibility - old keyword.
-    if sym_pos:
-        assume_a = 'pos'
+    # Get the correct lamch function.
+    # The LAMCH functions only exists for S and D
+    # So for complex values we have to convert to real/double.
+    if a1.dtype.char in 'fF':  # single precision
+        lamch = get_lapack_funcs(('lamch',), (np.array(0, dtype='f'),))[0]
+    else:
+        lamch = get_lapack_funcs(('lamch',), (np.array(0, dtype='d'),))[0]
+
+    # Currently we do not have the other forms of the norm calculators
+    #   lansy, lanpo, lanhe.
+    # However, in any case they only reduce computations slightly...
+    lange = get_lapack_funcs(('lange',), (a1,))[0]
+    # Since the I-norm and 1-norm are the same for symmetric matrices
+    # we can collect them all in this one call
+    # Note however, that when issuing 'gen' and form!='none', then
+    # the I-norm should be used
+    if transposed:
+        trans = 1
+        norm = 'I'
+        if r_or_c is not float:
+            raise ValueError('scipy.linalg.solve can currently not solve '
+                             'a^T x = b or a^H x = b for complex matrices.')
+    else:
+        trans = 0
+        norm = '1'
+    anorm = lange(norm, a1)
```

Comment:
Empty space nitpicking: empty line after end of blocks (here, before `anorm = ...`).
Empty line before comment block (e.g. after `lange = ...` above).
Generally no empty line not before `else` and `elif` (below).

Reply:
I am unsure how to interpret your last phrase, should I remove the empty lines just above `elif` and `else` in the "big" block of comparisons?

Reply:
@pv I have removed the spaces. I have looked in decomp.py and decomp_qr.py and done as there (no spaces, comments above).